src/topology/category/Profinite/cofiltered_limit.lean

```diff
@@ -110,4 +114,129 @@ begin
+    locally_constant.unflip (locally_constant.comap (C.π.app j0) ggg).flip, by simp,
+  rw this, clear this,
+  congr' 1,
+  ext1 a,
+  change ff a = _,
+  rw h,
+  dsimp [ggg, gg],
+  ext1,
+  repeat {
+    rw locally_constant.coe_comap,
+    dsimp [locally_constant.flip, locally_constant.unflip] },
+  { congr' 1,
+    change _ = ((C.π.app j0) ≫ (F.map (fs a))) x,
+    rw C.w },
+  all_goals { continuity },
+end
+
+theorem exists_locally_constant_fintype_nonempty {α : Type*} [fintype α] [nonempty α]
+  (f : locally_constant C.X α) :
+  ∃ (j : J) (g : locally_constant (F.obj j) α), f = g.comap (C.π.app _) :=
+begin
+  inhabit α,
+  obtain ⟨j,gg,h⟩ := exists_locally_constant_fintype_aux _ hC f,
+  let ι : α → α → fin 2 := λ a b, if a = b then 0 else 1,
+  let σ : (α → fin 2) → α := λ f, if h : ∃ (a : α), ι a = f then h.some else arbitrary _,
+  refine ⟨j, gg.map σ, _⟩,
+  ext,
+  rw locally_constant.coe_comap _ _ (C.π.app j).continuous,
+  dsimp [σ],
+  have h1 : ι (f x) = gg (C.π.app j x),
+  { change f.map (λ a b, if a = b then (0 : fin 2) else 1) x = _,
+    rw [h, locally_constant.coe_comap _ _ (C.π.app j).continuous] },
+  have h2 : ∃ a : α, ι a = gg (C.π.app j x) := ⟨f x, h1⟩,
+  rw dif_pos h2,
+  apply_fun ι,
+  { rw h2.some_spec,
+    exact h1 },
+  { intros a b hh,
+    apply_fun (λ e, e a) at hh,
+    dsimp [ι] at hh,
+    rw if_pos rfl at hh,
+    split_ifs at hh with hh1 hh1,
+    { exact hh1.symm },
+    { exact false.elim (bot_ne_top hh) } }
+end
+
+/-- Any locally constant function from a cofiltered limit of profinite sets factors through
+one of the components. -/
+theorem exists_locally_constant {α : Type*} (f : locally_constant C.X α) :
+  ∃ (j : J) (g : locally_constant (F.obj j) α), f = g.comap (C.π.app _) :=
+begin
+  let S := f.discrete_quotient,
+  let ff : S → α := f.lift,
+  by_cases hα : nonempty S,
+  { resetI,
+    let f' : locally_constant C.X S := ⟨S.proj, S.proj_is_locally_constant⟩,
+    obtain ⟨j,g',h⟩ := exists_locally_constant_fintype_nonempty _ hC f',
+    use j,
+    refine ⟨⟨ff ∘ g', g'.is_locally_constant.comp _⟩,_⟩,
+    ext1 t,
+    apply_fun (λ e, e t) at h,
+    rw locally_constant.coe_comap _ _ (C.π.app j).continuous at h ⊢,
+    dsimp at h ⊢,
+    rw ← h,
+    refl },
+  { suffices : ∃ j : J, ¬ nonempty (F.obj j),
+    { obtain ⟨j,hj⟩ := this,
+      use j,
+      refine ⟨⟨λ x, false.elim (hj ⟨x⟩), λ A, _⟩, _⟩,
+      { convert is_open_empty,
+        rw set.eq_empty_iff_forall_not_mem,
+        intros x,
+        exact false.elim (hj ⟨x⟩) },
+      { ext x,
+        exact false.elim (hj ⟨C.π.app j x⟩) } },
+    rw ← not_forall,
+    intros h,
+    apply hα,
+    haveI : ∀ j : J, nonempty ((F ⋙ Profinite_to_Top).obj j) := h,
+    haveI : ∀ j : J, t2_space ((F ⋙ Profinite_to_Top).obj j) := λ j,
+      (infer_instance : t2_space (Profinite_to_Top.obj _)),
+    haveI : ∀ j : J, compact_space ((F ⋙ Profinite_to_Top).obj j) := λ j,
+      (infer_instance : compact_space (Profinite_to_Top.obj _)),
```

Comment:
Does it work to make this `infer_instance : compact_space (F.obj j)` rather than giving this specific form and also making the new instances?

Reply:
This will probably work. I'll try it out.

Reply:
Yep, that worked, and I also removed the new instances.